### PR TITLE
feat: Add C# language support

### PR DESCRIPTION
**Pull request by urbit-pilled**
/claim #438 

PS: The `tree-sitter-c-sharp` submodule has introduced a new `tree-sitter-language` crate (https://crates.io/crates/tree-sitter-language) to decouple grammar parsing with the rest of tree-sitter. Unfortunately this is incompatible with `tree-sitter-facade` so I reverted the `tree-sitter-language` changes.

This is why the submodule doesn't point to the official `tree-sitter-c-sharp` repository.

### CODE REVIEW

File: crates/core/src/test.rs [hunk 1]
```diff
@@ -15305,3 +15305,789 @@ fn or_file() {
...
+            pattern: r#"
+                |language csharp
+                |
+                |`public class $name {
+                |    public $name($params) {
+                |        $assignments
+                |    }
+                |
+                |    $_
+                |    $_
```
Comment (morgante) [inline crates/core/src/test.rs:15773]: For future tests we usually want to be a bit less strict on shape. I'd personally want to see `$...` work here.

File: crates/core/src/test.rs [hunk 2]
```diff
@@ -15305,3 +15305,789 @@ fn or_file() {
...
+fn csharp_groupby_keyword() {
+    run_test_expected({
+        TestArgExpected {
+            pattern: r#"
+                |language csharp
+                |
+                |`from $item in $collection
+                |group $item by $item.$key` =>
+                |                    `from $item in $collection
+                |                    group $item by $item.$key into g
```
Comment (morgante) [inline crates/core/src/test.rs:15997]: We should also add tests for specific snippets *inside* a linq query.

Ex.

```
language csharp

`group $item by $item.key` => .
```

**Review by morgante — APPROVED** [review]
Looks good, would just love a few more tests to award the bounty. Also please run this on a real C# repo and confirm it works.

**Comment by urbit-pilled** [comment]
Thanks for the review @morgante.
I'll add more tests,  test it out on a real repo and add tests within LINQ queries when I get the time (hopefully later today or tomorrow)

File: crates/core/src/test.rs (hunk 1 above)
Comment (urbit-pilled) [reply crates/core/src/test.rs:15773]: fixed in cee3e5f98f16f83de3242e944c966dbccfe7f26a

File: crates/core/src/test.rs (hunk 2 above)
Comment (urbit-pilled) [reply crates/core/src/test.rs:15997]: fixed in 7ed9e1f30cfd65bc9fb4b5dd6f99b43b4c3300ee

**Pull request merged**